> gsematrix
$href
$href$`GSE1133-GPL1073_series_matrix.txt.gz`
ExpressionSet (storageMode: lockedEnvironment)
assayData: 31373 features, 122 samples 
  element names: exprs 
protocolData: none
phenoData
  sampleNames: GSM18584 GSM18585 ... GSM18705 (122 total)
  varLabels: title geo_accession ... data_row_count (27 total)
  varMetadata: labelDescription
featureData
  featureNames: AFFX-18SRNAMur/X00686_3_at AFFX-18SRNAMur/X00686_5_at ... gnf1m31309_s_at (31373 total)
  fvarLabels: ID Gene title ... Platform_SEQUENCE (22 total)
  fvarMetadata: Column Description labelDescription
experimentData: use 'experimentData(object)'
Annotation: GPL1073 

$href$`GSE1133-GPL1074_series_matrix.txt.gz`
ExpressionSet (storageMode: lockedEnvironment)
assayData: 11391 features, 158 samples 
  element names: exprs 
protocolData: none
phenoData
  sampleNames: GSM18706 GSM18707 ... GSM18863 (158 total)
  varLabels: title geo_accession ... data_row_count (34 total)
  varMetadata: labelDescription
featureData
  featureNames: gnf1h00001_x_at gnf1h00002_x_at ... gnf1h11406_at (11391 total)
  fvarLabels: ID Gene title ... Platform_SEQUENCE (22 total)
  fvarMetadata: Column Description labelDescription
experimentData: use 'experimentData(object)'
Annotation: GPL1074 

$href$`GSE1133-GPL96_series_matrix.txt.gz`
ExpressionSet (storageMode: lockedEnvironment)
assayData: 22283 features, 158 samples 
  element names: exprs 
protocolData: none
phenoData
  sampleNames: GSM18865 GSM18866 ... GSM19022 (158 total)
  varLabels: title geo_accession ... relation (35 total)
  varMetadata: labelDescription
featureData
  featureNames: 1007_s_at 1053_at ... AFFX-TrpnX-M_at (22283 total)
  fvarLabels: ID Gene title ... GO:Component ID (21 total)
  fvarMetadata: Column Description labelDescription
experimentData: use 'experimentData(object)'
Annotation: GPL96 


$plung1
$plung1$GSE13134_series_matrix.txt.gz
ExpressionSet (storageMode: lockedEnvironment)
assayData: 55488 features, 35 samples 
  element names: exprs 
protocolData: none
phenoData
  sampleNames: GSM328803 GSM328804 ... GSM328837 (35 total)
  varLabels: title geo_accession ... data_row_count (45 total)
  varMetadata: labelDescription
featureData
  featureNames: 100001.1 100001.2 ... 332368.2 (55488 total)
  fvarLabels: ID BLOCK ... SPOT_ID (11 total)
  fvarMetadata: Column Description labelDescription
experimentData: use 'experimentData(object)'
Annotation: GPL6173 

# > pData(gsematrix$plung1[[1]])[1,]
#                                    title geo_accession                status submission_date last_update_date type
# GSM328803 Lung_healthy_control_Animal-20     GSM328803 Public on Nov 24 2009     Oct 09 2008      Nov 24 2009  RNA
#           channel_count                source_name_ch1 organism_ch1
# GSM328803             2 Lung_healthy_control_Animal-20   Sus scrofa
#                                                                                                                                                                                                     characteristics_ch1
# GSM328803 Breed: Danish Landrace/Yorkshire/Duroc cross, Herd: high health free from A. pleuropneumoniae, Gender: castrated males, Age: 8-10 weeks, Treatment: healthy control, Sample: Lung_healthy_control, Animal: 20
# 
# treatment_protocol_ch1
# GSM328803 Pigs assigned as healthy non-inoculated control animals were killed and sampled prior to inoculation of the infected group. Samples of approximately 500 mg tissue were taken from the lungs, snap frozen in liquid nitrogen and stored at minus 80 degrees until further use. ...
#           molecule_ch1                                                          extract_protocol_ch1 label_ch1
# GSM328803    total RNA RNeasy Midi Kit with DNase treatment following the enclosed protocol (Qiagen) Alexa-555
#                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         label_protocol_ch1
# GSM328803 20 Âµg total RNA was labelled by using the Superscript Direct cDNA Labeling System (Invitrogen) following the enclosed protocols. Spike-in RNA from the Lucidea Universal ScoreCard (Amersham Biosciences) was added to the labelling reactions. Green spike-in RNA was added to the common reference samples and red spike-in RNA was added to the samples. The fluorescently labelled cDNA was purified using the NucleoSpin 96 Extract II PCR Clean-up kit (Macherey-Nagel, DÃ¼ren, Germany).
#           taxid_ch1                                               source_name_ch2 organism_ch2
# GSM328803      9823 Reference, pool of equal amount of total-RNA from all samples   Sus scrofa
#                                                                                                                                                                                                                 characteristics_ch2
# GSM328803 Breed: Danish Landrace/Yorkshire/Duroc cross, Herd: high health free from A. pleuropneumoniae, Gender: castrated males, Age: 8-10 weeks, Reference, pool of equal amount of total-RNA from all lung samples in experiment
#                                                          biomaterial_provider_ch2
# GSM328803 Technical University of Denmark, National Veterinary Institute, Denmark
#                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          treatment_protocol_ch2
# GSM328803 Pigs assigned as healthy non-inoculated control animals were killed and sampled prior to inoculation of the infected group. Samples of approximately 500 mg tissue were taken from the lungs, snap frozen in liquid nitrogen and stored at minus 80 degrees until further use. ... All animal procedures were approved by the Danish Animal Experiments Inspectorate. Immediately after killing, lungs were inspected for sizes and numbers of pulmonary lesions. ...
#           molecule_ch2                                                          extract_protocol_ch2 label_ch2
# GSM328803    total RNA RNeasy Midi Kit with DNase treatment following the enclosed protocol (Qiagen) Alexa-647
#                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         label_protocol_ch2
# GSM328803 20 Âµg total RNA was labelled by using the Superscript Direct cDNA Labeling System (Invitrogen) following the enclosed protocols. Spike-in RNA from the Lucidea Universal ScoreCard (Amersham Biosciences) was added to the labelling reactions. Green spike-in RNA was added to the common reference samples and red spike-in RNA was added to the samples. The fluorescently labelled cDNA was purified using the NucleoSpin 96 Extract II PCR Clean-up kit (Macherey-Nagel, DÃ¼ren, Germany). The labelled reference samples were mixed and divided into aliquots before combining it with a labelled sample.
#           taxid_ch2
# GSM328803      9823
#                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           hyb_protocol
# GSM328803 The slides were hybridized in a Discovery XT hybridization station (Ventana Discovery Systems, Tucson, AZ, USA). Transfer Chip Prep-2 from 4 ÂºC to room temperature 1 hour before use. Prepare ChipSpread by mixing equal volumes of ChipSpread A (20 mg/mL BSA, 4x SSC, 0.5 mg/mL sodium azide) and B (formamide; 2 mg/mL SDS) and incubate at room temperature for 1 hour before use. A total of 2.5 mL is needed per slide. Print labels, trim them and place them on the slides. Mix the Chip Map reagents (Chip Prep-1, -2 and - 3) by inversion, remove the cap and place the reagents in the Discovery. Place the slides in the machine and initiate the run. Cover slide with 2.5 mL ChipSpread when the message appears (after few minutes). The machine now runs for app. 1.5 hours to pre-hybridize the slides. Heat a waterbath to 90Â°C or use a PCR machine. Mix the Chiphybe80, add 200 ÂµL to the sample (<20 ÂµL) and mix carefully. Heat the sample mixture at 90Â°C for 3 minutes and mix carefully by pipetting. Press the button on the machine which then prepares the slides for hybridization. When the message appears apply the samples onto the slides and press the button and the machine hybridizes at 48 ÂºC for 6 hours. Wipe oil from backside of slides using a clean-room napkin and place slides in the slide-holder from the High Throughput Wash Station (Telechem, cat.no. HTW) placed in a mTub filled with RiboWash. If processing more than 20 slides, place equal number of slides in two slide-holders and continue in parallel. Transfer the slide-holder to a HTW filled with RiboWash and wash for 2 min with magnetic stirring at 700 rpm. Refill the HTW with RiboWash and repeat the wash. Dip the slide-holder in 2x SSC filled in a mTub (200 mL 20x SSC, Elga H2O + 1800 mL water). Transfer the slide-holder to a HTW filled with 2x SSC and wash for 2 min with magnetic stirring at 700 rpm. Refill the HTW with 2x SSC and repeat the wash. Dip the slide-holder 10 times in 0.1x SSC filled in a mTub (5 mL 20x SSC, Elga H2O + 995 mL water) and leave the holder submerged in 0.1x SSC. Transfer the slides to a mBox slide holder placed in a mTub filled with Elga H2O. Dry arrays by centrifugation (at 300 x g for 4 min placed in a mBox).
#                                                                                                                                                                                                                                                                                                                           scan_protocol
# GSM328803 Scanner: ScanArray Express HT system (Perkin Elmer), 5 Âµm resolution, 100 % laser power and PMT adjusted individually for each channel. Image analysis software: GenePix Pro (version 6.0.1.27, Molecular Devices) using FeatureType = Irregular Filled and MorphologicalClosingFollowedByOpening as background measurement.
#                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   description
# Samples of lung tissue were furthermore sampled from healthy non-inoculated control pigs from the same heard. Samples were investigated for changes in gene expression by means of quantitative real-time PCR and global cDNA microarrays.
#                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 data_processing
# GSM328803 Statistical analysis was carried out in the R computing environment (version 2.6.1 for Windows) using the package Linear Models for Microarray Analysis (Limma, version 2.12.0) which is part of the Bioconductor project. Control spots and spots with more than 50 % of saturated pixels were weighted 0 before the log2-transformed ratios of Alexa-555 to Alexa-647 (not background corrected) were normalized within-slide using global-loess with default parameters as implemented in Limma. The set of normalized log-ratios were analyzed in Limma to identify genes being significantly differentially expressed between lungs of uninfected control animals and the three different regions sampled from lungs of infected animals (unaffected areas, the demarcation zone, and necrotic areas). The false discovery rate was controlled using the method of Benjamini and Hochberg as implemented in Limma and a corrected P-value below 0.01 was considered significant.
#           platform_id     data_row_count
# GSM328803     GPL6173              55488

plung1.controls <- str_detect(pData(gsematrix$plung1[[1]])$title, "control")
gsematrix$plung1[[1]] <- gsematrix$plung1[[1]][, plung1.controls]

$plung2
$plung2$GSE28871_series_matrix.txt.gz
ExpressionSet (storageMode: lockedEnvironment)
assayData: 43603 features, 12 samples 
  element names: exprs 
protocolData: none
phenoData
  sampleNames: GSM714777 GSM714778 ... GSM714788 (12 total)
  varLabels: title geo_accession ... data_row_count (33 total)
  varMetadata: labelDescription
featureData
  featureNames: A_72_P000001 A_72_P000006 ... A_72_P438059 (43603 total)
  fvarLabels: ID Common name ... DESCRIPTION (7 total)
  fvarMetadata: Column Description labelDescription
experimentData: use 'experimentData(object)'
Annotation: GPL10162 

# > pData(gsematrix$plung2[[1]])[1,]
#                           title geo_accession                status submission_date last_update_date type channel_count
# GSM714777 Pig-control-3day-rep1     GSM714777 Public on Apr 27 2011     Apr 26 2011      Apr 28 2011  RNA             1
#                  source_name_ch1 organism_ch1 characteristics_ch1 characteristics_ch1.1
# GSM714777 Lung,control,3day,rep1   Sus scrofa        tissue: Lung   disease state: none
#                                                                                                                                          treatment_protocol_ch1
# GSM714777 The animals were sacrificed to collect lung tissues, which were immediately preserved in liquid nitrogen and subsequently stored in a -80â„ƒ freezer.
#                                                                                                                                                                                                                                                                                           growth_protocol_ch1
# GSM714777 A total of 30 piglets of high-health status were obtained from a commercial herd and were raised in isolated facilities.Prior to challenge, all pigs sera  were confirmed negative for the antibodies against H1 and H3 swine influenza viruses by HI (Hemagglutination Inhibition) Test and ELISA.
#           molecule_ch1
# GSM714777    total RNA
#                                                                                                                                                                                                                                                                                                                                                                                                                                        extract_protocol_ch1
# GSM714777 RNA extraction from pig lungs prepared from the same lesion, areas of localized infection (with viral mRNA present) was performed using TRIzol following standard instructions (Invitrogen, Carlsbad, CA) and a clean-up was carried out using RNeasy columns (Qiagen, Beijing, China). RNA integrity and concentration were evaluated by Agilent 2100 Bioanalyzer followed by manufacturer instructions (Agilent Technologies, Palo Alto, CA).
#           label_ch1
# GSM714777       Cy3
#                                                                                                                                                                                                                                                                                                                                                          label_protocol_ch1
# GSM714777 Cyanine-3 (Cy3) labeled cRNA was prepared from 0.5 ug RNA using the One-Color Low RNA Input Linear Amplification PLUS kit (Agilent) according to the manufacturer instructions, followed by RNAeasy column purification (QIAGEN, Valencia, CA).  Dye incorporation and cRNA yield were checked with the NanoDrop ND-1000 Spectrophotometer(Shanghaibio, China).
#           taxid_ch1
# GSM714777      9823
#                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          hyb_protocol
# GSM714777 0.875 ug of Cy3-labelled cRNA (specific activity >10.0 pmol Cy3/ug cRNA) was fragmented at 60Â°C for 30 minutes in a reaction volume of 55ul containing 25x Agilent fragmentation buffer and 10x Agilent blocking agent following the manufacturers instructions. On completion of the fragmentation reaction, 55ul of 2x Agilent hybridization buffer was added to the fragmentation mixture and hybridized to Agilent 4x44K Porcine Oligo Microarraysï¼‰ for 17 hours at 65 Â°C in a rotating Agilent hybridization oven.  After hybridization, microarrays were washed 1 minute at room temperature with GE Wash Buffer 1 (Agilent) and 1 minute with 37 Â°C GE Wash buffer 2 (Agilent), then dried immediately by brief centrifugation.
#                                                                                                                                                scan_protocol
# GSM714777 Slides were scanned immediately after washing on the Agilent DNA Microarray Scanner (G2565BA) using one color scan setting for 4x44k array slides.
#                                                                                                                                                             description
# GSM714777 We only selected 3 pieces of lung tissues of 3 different pigs from each pool of control 3 day, infected 3 day, control 7 day and infected 7 day respectively.
#                                                                                                                  data_processing
# GSM714777 Agilent Feature Extraction was used for reading the data, Scan resolution 5Î¼m, PMT 100%, 10% and LOWESS normalization
#           platform_id contact_name     contact_email
# GSM714777    GPL10162  yongtao,,Li yongtaole@126.com
#                                                                     contact_institute          contact_address
# GSM714777 Unit of Animal Infectious Diseases, National Key Laboratory of Agricultural shizisan street,number 1
#           contact_city contact_state contact_zip/postal_code contact_country
# GSM714777        wuhan         Hubei                  430070           China
#                                                                                           supplementary_file
# GSM714777 ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/supplementary/samples/GSM714nnn/GSM714777/GSM714777.txt.gz
#           data_row_count
# GSM714777          43603

plung2.controls <- str_detect(pData(gsematrix$plung2[[1]])$title, "control")
gsematrix$plung2[[1]] <- gsematrix$plung2[[1]][, plung2.controls]


$plung3
$plung3$GSE12194_series_matrix.txt.gz
ExpressionSet (storageMode: lockedEnvironment)
assayData: 13677 features, 112 samples 
  element names: exprs 
protocolData: none
phenoData
  sampleNames: GSM306300 GSM306309 ... GSM306830 (112 total)
  varLabels: title geo_accession ... data_row_count (44 total)
  varMetadata: labelDescription
featureData
  featureNames: 1 2 ... 13677 (13677 total)
  fvarLabels: ID oligo_id ... TIGR_TC_ID (5 total)
  fvarMetadata: Column Description labelDescription
experimentData: use 'experimentData(object)'
Annotation: GPL1881 

# > pData(gsematrix$plung3[[1]])[1,]
#             title geo_accession                status submission_date last_update_date type channel_count source_name_ch1
# GSM306300 Lung A1     GSM306300 Public on Dec 31 2008     Jul 18 2008      Jul 22 2008  RNA             2  Lung A1 source
#           organism_ch1 characteristics_ch1 characteristics_ch1.1 characteristics_ch1.2
# GSM306300   Sus scrofa       Tissue: Lung;          Line: Index;   Viremia Class: Low;
#                                           characteristics_ch1.3 molecule_ch1
# GSM306300 Infection Status: Infected with PRRSV strain 97-7985     total RNA
#                                          extract_protocol_ch1 label_ch1
# GSM306300 RNAs were extracted using Trizol-chlorofrom method.       Cy5
#                                                                                                                                                                                                                          label_protocol_ch1
# GSM306300 Two Âµg of total RNA was reverse-transcribed using oligo-DT primer and reagents from the Ambion Amino Allyl MessageAmp aRNA Amplification Kit (Ambion, Austin, TX). Following cRNA generation, 5 Âµg of cRNA was coupled to CY5. 
#           taxid_ch1       source_name_ch2 organism_ch2
# GSM306300      9823 Lung Reference Sample   Sus scrofa
#                                                                                                                                                                                                                                                                                                                  characteristics_ch2
# GSM306300 The reference sample was developed from four uninfected control full sibling littermates of infected pigs that had a PC value of 0 and that were not included in the sample of 28 infected pigs used in the microarray experiment.  The reference sample consisted of two pigs from Line Index and two pigs from Line HD. 
#           characteristics_ch2.1 molecule_ch2                                extract_protocol_ch2 label_ch2
# GSM306300          Tissue: Lung    total RNA RNAs were extracted using Trizol-chlorofrom method.       Cy3
#                                                                                                                                                                                                                         label_protocol_ch2
# GSM306300 Two Âµg of total RNA was reverse-transcribed using oligo-DT primer and reagents from the Ambion Amino Allyl MessageAmp aRNA Amplification Kit (Ambion, Austin, TX). Following cRNA generation, 5 Âµg of cRNA was coupled to CY3.
#           taxid_ch2
# GSM306300      9823
#                                                                                                                                                                                      hyb_protocol
# GSM306300 The samples were hybridized overnight using Ambion slide hyb buffer at 42ÂºC for 16 hours. Slides were washed per manufacturersâ€™ suggestions and scanned with an Axon 4000B scanner. 
#                                                 scan_protocol description
# GSM306300 Images were quantified using an Axon 4000B scanner.         N/A
#                                                                                                                                                                                                                            data_processing
# GSM306300 Global Median Normalization was performed.  Data points (F635 Median - B635 and F532 Median - B532) falling in the range of -150 to 150 in intensity value and those with ratios less than -15 and greater than 15 were removed.
#           platform_id    contact_name             contact_email  contact_phone contact_department      contact_institute
# GSM306300     GPL1881 Rodger,,Johnson rkjohnso@unlnotes.unl.edu (402) 472-6404     Animal Science University of Nebraska
#                                                              contact_address contact_city contact_state
# GSM306300 University of Nebraska East Campus Loop Animal Science Bldg. A218i      Lincoln            NE
#           contact_zip/postal_code contact_country supplementary_file data_row_count
# GSM306300              68583-0908             USA               NONE          13677

plung3.controls <- str_detect(pData(gsematrix$plung3[[1]])$characteristics_ch1, "Lung") & str_detect(pData(gsematrix$plung3[[1]])$characteristics_ch1, "Uninfected")
gsematrix$plung3[[1]] <- gsematrix$plung3[[1]][, plung3.controls]

$plung4
$plung4$GSE2335_series_matrix.txt.gz
ExpressionSet (storageMode: lockedEnvironment)
assayData: 13677 features, 24 samples 
  element names: exprs 
protocolData: none
phenoData
  sampleNames: GSM43151 GSM43152 ... GSM43174 (24 total)
  varLabels: title geo_accession ... data_row_count (36 total)
  varMetadata: labelDescription
featureData
  featureNames: 1 2 ... 13677 (13677 total)
  fvarLabels: ID oligo_id ... TIGR_TC_ID (5 total)
  fvarMetadata: Column Description labelDescription
experimentData: use 'experimentData(object)'
Annotation: GPL1881 

# > pData(gsematrix$plung4[[1]])[1,]
#                  title geo_accession                status submission_date last_update_date type channel_count
# GSM43151 Tissue slide1      GSM43151 Public on Mar 05 2005     Feb 23 2005      Oct 28 2005  RNA             2
#          source_name_ch1 organism_ch1 molecule_ch1 taxid_ch1         source_name_ch2 organism_ch2 molecule_ch2 taxid_ch2
# GSM43151    Pig 436 Lung   Sus scrofa    total RNA      9823 Pig 436 Small Intestine   Sus scrofa    total RNA      9823
#                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    description
# GSM43151 Total RNA from different porcine tissues, lung, muscle (ham), liver, and small intestine (duodenum; approximately 5 cm from stomach), were isolated by RNeasy kit and treated by DNaseI. For generating amino-allyl-dUTP labeled single strand cDNA, 30 ug RNA from the four tissue samples were reverse transcribed for 2-3 h at 42C in the presence of 6 ug random and 1 ug oligo-dT(18) primers (Invitrogen/Life Technologies), 1 X first strand buffer, 10mM DTT, dNTP (25 mM dCTP, 25 mM dGTP, 25 mM dATP, 15 mM dTTP, 10 mM aa-dUTP), and 400 units of Superscript II reverse transcriptase (Invitrogen/Life Technologies).  RNA was hydrolyzed with 1M NaOH, and then neutralized with 1M HEPES (pH 7.0). First-strand cDNA was precipitated, air dried and re-suspended with 4 ul 0.1M Na2CO3 buffer (pH 9.0), mixed with either Cy3 or Cy5 NHS-ester (Amersham Pharmacia), and incubated 1 hr at room temperature. Cy3 and Cy5 labeled cDNA were purified on QIAquick PCR purification columns (Qiagen, Valencia, CA) according to manufacturerâ€™s instructions, except that QIAquick wash buffer was replaced by phosphate buffer.  Eluted cDNA were precipitated by ETOH and hybridized to the Qiagen-NRSP-8 porcine oligonucleotide microarray at 42C for 12-16 hr in hybridization solution containing 50% formamide, 5 X SSC, 0.1% SDS and 0.2 ug/ul sheared salmon sperm DNA.  After hybridization, microarray slides were washed by 2 X SSC, 0.1% SDS for 5 min at 42C, 0.1 X SSC for 1 min at room temperature for four times, and 0.01 X SSC for 10 sec at room temperature, and dried by centrifugation.Slides were scanned at 10 um resolution using a ScanArray 5000 scanner under conditions to limit saturation to <1% and were saved as TIFF images. The intensities of spots on each image were quantified by ImaGene 5.1 software, and data were saved as .txt files for further analyses.
#           description.1         description.2     description.3              description.4 description.5 platform_id
# GSM43151 Keywords = pig Keywords = microarray Keywords = tissue Keywords = gene expression                   GPL1881
#                  contact_name        contact_email contact_phone  contact_fax contact_laboratory contact_department
# GSM43151 Christopher,K,Tuggle cktuggle@iastate.edu  515-294-4252 515-294-2401 Molecular Genetics     Animal Science
#              contact_institute  contact_address contact_city contact_state contact_zip/postal_code contact_country
# GSM43151 Iowa State University 2255 Kildee Hall         Ames            IA                   50011             USA
#          supplementary_file data_row_count
# GSM43151               NONE          13677

plung4.controls <- str_detect(pData(gsematrix$plung4[[1]])$source_name_ch1, "lung") | str_detect(pData(gsematrix$plung4[[1]])$source_name_ch2, "lung")
gsematrix$plung4[[1]] <- gsematrix$plung4[[1]][, plung4.controls]

$plung5
$plung5$GSE2339_series_matrix.txt.gz
ExpressionSet (storageMode: lockedEnvironment)
assayData: 13677 features, 9 samples 
  element names: exprs 
protocolData: none
phenoData
  sampleNames: GSM43176 GSM43177 ... GSM43228 (9 total)
  varLabels: title geo_accession ... data_row_count (37 total)
  varMetadata: labelDescription
featureData
  featureNames: 1 2 ... 13677 (13677 total)
  fvarLabels: ID oligo_id ... TIGR_TC_ID (5 total)
  fvarMetadata: Column Description labelDescription
experimentData: use 'experimentData(object)'
Annotation: GPL1881 

# > pData(gsematrix$plung5[[1]])[1,]
#                title geo_accession                status submission_date last_update_date type channel_count
# GSM43176 Lung slide1      GSM43176 Public on Mar 05 2005     Feb 23 2005      Oct 28 2005  RNA             2
#                          source_name_ch1 organism_ch1 molecule_ch1 taxid_ch1
# GSM43176 Day -3 control lung (Pig ID 19)   Sus scrofa    total RNA      9823
#                                                                 source_name_ch2 organism_ch2 molecule_ch2 taxid_ch2
# GSM43176 lung sample after 24 hour intranasal Salmonella infection  (pig ID 13)   Sus scrofa    total RNA      9823
#                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           description
# GSM43176 Total RNAs from control (3 pigs), 24 hour (3 pigs) and 48 hour (3 pigs) porcine lungs after intranasal Salmonella Choleraesuis infection were isolated by RNeasy kit and treated by DNaseI. For generating amino-allyl-dUTP labeled single strand cDNA, 20 ug RNA from the control and infected lungs were reverse transcribed for 2-3 h at 42C in the presence of 6 ug random and 1 ug oligo-dT(18) primers (Invitrogen/Life Technologies), 1 X first strand buffer, 10mM DTT, dNTP (25 mM dCTP, 25 mM dGTP, 25 mM dATP, 15 mM dTTP, 10 mM aa-dUTP), and 400 units of Superscript II reverse transcriptase (Invitrogen/Life Technologies). RNA was hydrolyzed with 1M NaOH, and then neutralized with 1M HEPES (pH 7.0). First-strand cDNA was precipitated, air dried and re-suspended with 4 ul 0.1M Na2CO3 buffer (pH 9.0), mixed with either Cy3 or Cy5 NHS-ester (Amersham Pharmacia), and incubated 1 hr at room temperature. Cy3 and Cy5 labeled cDNA were purified on QIAquick PCR purification columns (Qiagen, Valencia, CA) according to manufacturerâ€™s instructions, except that QIAquick wash buffer was replaced by phosphate buffer. Eluted cDNA were precipitated by ETOH and hybridized to the Qiagen-NRSP-8 porcine oligonucleotide microarray at 42C for 12-16 hr in hybridization solution containing 50% formamide, 5 X SSC, 0.1% SDS and 0.2 ug/ul sheared salmon sperm DNA. After hybridization, microarray slides were washed by 2 X SSC, 0.1% SDS for 5 min at 42C, 0.1 X SSC for 1 min at room temperature for four times, and 0.01 X SSC for 10 sec at room temperature, and dried by centrifugation.Slides were scanned at 10 um resolution using a ScanArray 5000 scanner under conditions to limit saturation to <1% and were saved as TIFF images. The intensities of spots on each image were quantified by ImaGene 5.1 software, and data were saved as .txt files for further analyses.
#           description.1         description.2   description.3              description.4        description.5
# GSM43176 Keywords = pig Keywords = microarray Keywords = lung Keywords = gene expression Keywords = infection
#          description.6 platform_id         contact_name        contact_email contact_phone  contact_fax contact_laboratory
# GSM43176                   GPL1881 Christopher,K,Tuggle cktuggle@iastate.edu  515-294-4252 515-294-2401 Molecular Genetics
#          contact_department     contact_institute  contact_address contact_city contact_state contact_zip/postal_code
# GSM43176     Animal Science Iowa State University 2255 Kildee Hall         Ames            IA                   50011
#          contact_country supplementary_file data_row_count
# GSM43176             USA               NONE          13677

plung5.controls <- str_detect(pData(gsematrix$plung5[[1]])$source_name_ch1, "control") | str_detect(pData(gsematrix$plung5[[1]])$source_name_ch2, "control")
gsematrix$plung5[[1]] <- gsematrix$plung5[[1]][, plung5.controls]


$hlung
$hlung$`GSE47460-GPL14550_series_matrix.txt.gz`
ExpressionSet (storageMode: lockedEnvironment)
assayData: 15261 features, 429 samples 
  element names: exprs 
protocolData: none
phenoData
  sampleNames: GSM1149948 GSM1149950 ... GSM1150547 (429 total)
  varLabels: title geo_accession ... data_row_count (42 total)
  varMetadata: labelDescription
featureData
  featureNames: A_23_P100001 A_23_P100022 ... A_32_P99902 (15261 total)
  fvarLabels: ID SPOT_ID ... SEQUENCE (17 total)
  fvarMetadata: Column Description labelDescription
experimentData: use 'experimentData(object)'
Annotation: GPL14550 

# > pData(gsematrix$hlung[[1]])[1,]
# ...                source_name_ch1 organism_ch1    characteristics_ch1 characteristics_ch1.1 characteristics_ch1.2
# GSM1149948 Flash frozen whole lung Homo sapiens disease state: Control           Sex: 1-Male               age: 75
#            characteristics_ch1.3  characteristics_ch1.4        characteristics_ch1.5       characteristics_ch1.6
# GSM1149948 gold stage: 0-At Risk smoker?: 2-Ever (>100) %predicted fev1 (pre-bd): 96 %predicted fvc (pre-bd): 97
#            characteristics_ch1.7 characteristics_ch1.8 characteristics_ch1.9 characteristics_ch1.10 characteristics_ch1.11
# GSM1149948   %predicted dlco: 78                                                                                          
#                                                                                                                                                                                                                                                                                  extract_protocol_ch1
# GSM1149948 30 mg of whole lung was homogenized with Trizol and then total RNA was extracted using the QiaCube system (Qiagen, Valencia CA).  RNA quality was determined by Agilent Bioanalyzer 2100 (Agilent Technologies Inc, Santa Clara, CA) with RIN > 7.0 being acceptable for microarray usage.
#                                                                                                                                                                                                                                                                                label_protocol_ch1
# GSM1149948 Cyanine-3 (Cy3) labeled cRNA was prepared from 0.5 ug RNA using the QuickAmp kit (Agilent) according to the manufacturer instructions, followed by RNAeasy column purification (QIAGEN).  Dye incorporation and cRNA yield were checked with the NanoDrop ND-1000 Spectrophotometer.
#                                                                                                                                                                                                                  hyb_protocol
# GSM1149948 1.5 ug of Cy3-labelled cRNA (specific activity >10.0 pmol Cy3/ug cRNA) was hybridized to the specified arrays as per manufacturer instructions.  Hybridization was in a rotating oven at the specified temperature
#                                                                                                                                                                  scan_protocol
# GSM1149948 Slides were scanned immediately after washing on the Agilent DNA Microarray Scanner (G2505B) using one color scan setting for 4x44K arrays or 8x60K arrays as noted
#                                                                                                                                                                                                                                                                                                                                                                                                                                                             data_processing
# GSM1149948 The scanned images were analyzed with Feature Extraction Software 9.1 (Agilent) using default perameters to obtain background subtracted and spatially detrended Processed Signal intensities.  Features flagged in Feature Extraction as Feature Non-uniform outliers were excluded.  The resulting array data was normalized using a cyclic loess approach in pairwise fashion.  Probes were then collapsed to one gene/one probe by highest signal intensity.
#            platform_id data_row_count
#               GPL14550          15261

hlung.controls <- gsematrix$hlung[[1]]$characteristics_ch1 == "disease state: Control"
gsematrix$hlung[[1]] <- gsematrix$hlung[[1]][,hlung.controls]

$hlung$`GSE47460-GPL6480_series_matrix.txt.gz`
ExpressionSet (storageMode: lockedEnvironment)
assayData: 15261 features, 153 samples 
  element names: exprs 
protocolData: none
phenoData
  sampleNames: GSM1149949 GSM1149953 ... GSM1150360 (153 total)
  varLabels: title geo_accession ... data_row_count (42 total)
  varMetadata: labelDescription
featureData
  featureNames: A_23_P100001 A_23_P100022 ... A_32_P99902 (15261 total)
  fvarLabels: ID SPOT_ID ... SEQUENCE (17 total)
  fvarMetadata: Column Description labelDescription
experimentData: use 'experimentData(object)'
Annotation: GPL6480 

# > pData(gsematrix$hlung[[2]])[1,]
#                      title geo_accession                status submission_date last_update_date type channel_count
# GSM1149955 LT009615LU_CTRL    GSM1149955 Public on May 30 2013     May 29 2013      May 30 2013  RNA             1
#                    source_name_ch1 organism_ch1    characteristics_ch1 characteristics_ch1.1 characteristics_ch1.2
# GSM1149955 Flash frozen whole lung Homo sapiens disease state: Control         Sex: 2-Female               age: 49
#            characteristics_ch1.3  characteristics_ch1.4         characteristics_ch1.5        characteristics_ch1.6
# GSM1149955 gold stage: 0-At Risk smoker?: 2-Ever (>100) %predicted fev1 (pre-bd): 105 %predicted fvc (pre-bd): 119
#                     characteristics_ch1.7         characteristics_ch1.8 characteristics_ch1.9 characteristics_ch1.10
# GSM1149955 %predicted fev1 (post-bd): 112 %predicted fvc (post-bd): 119  %predicted dlco: 102                       

hlung2.controls <- gsematrix$hlung[[2]]$characteristics_ch1 == "disease state: Control"
gsematrix$hlung[[2]] <- gsematrix$hlung[[2]][,hlung2.controls]

$palmac
$palmac$GSE45145_series_matrix.txt.gz
ExpressionSet (storageMode: lockedEnvironment)
assayData: 47845 features, 140 samples 
  element names: exprs 
protocolData: none
phenoData
  sampleNames: GSM1097889 GSM1097890 ... GSM1098028 (140 total)
  varLabels: title geo_accession ... data_row_count (36 total)
  varMetadata: labelDescription
featureData
  featureNames: AFFX-BioB-3_at AFFX-BioB-5_at ... ssc-mir-92a-2_st (47845 total)
  fvarLabels: ID Probeset ... GO_MF (15 total)
  fvarMetadata: Column Description labelDescription
experimentData: use 'experimentData(object)'
Annotation: GPL16569 


$halmac
$halmac$GSE13896_series_matrix.txt.gz
ExpressionSet (storageMode: lockedEnvironment)
assayData: 54675 features, 70 samples 
  element names: exprs 
protocolData: none
phenoData
  sampleNames: GSM219241 GSM219242 ... GSM349958 (70 total)
  varLabels: title geo_accession ... relation (37 total)
  varMetadata: labelDescription
featureData
  featureNames: 1007_s_at 1053_at ... AFFX-TrpnX-M_at (54675 total)
  fvarLabels: ID Gene title ... GO:Component ID (21 total)
  fvarMetadata: Column Description labelDescription
experimentData: use 'experimentData(object)'
Annotation: GPL570 


$ppbmc1
$ppbmc1$GSE17320_series_matrix.txt.gz
ExpressionSet (storageMode: lockedEnvironment)
assayData: 19200 features, 28 samples 
  element names: exprs 
protocolData: none
phenoData
  sampleNames: GSM433321 GSM433322 ... GSM433348 (28 total)
  varLabels: title geo_accession ... data_row_count (43 total)
  varMetadata: labelDescription
featureData
  featureNames: Alien1 Alien1.2 ... uYIR3.4 (19200 total)
  fvarLabels: ID Block ... SPOT_ID (9 total)
  fvarMetadata: Column Description labelDescription
experimentData: use 'experimentData(object)'
Annotation: GPL7151 


$ppbmc2
$ppbmc2$GSE23503_series_matrix.txt.gz
ExpressionSet (storageMode: lockedEnvironment)
assayData: 24123 features, 22 samples 
  element names: exprs 
protocolData: none
phenoData
  sampleNames: GSM575712 GSM575713 ... GSM575733 (22 total)
  varLabels: title geo_accession ... data_row_count (43 total)
  varMetadata: labelDescription
featureData
  featureNames: AFFX-BioB-3_at AFFX-BioB-5_at ... SscAffx.9.1.S1_at (24123 total)
  fvarLabels: ID Gene title ... GO:Component ID (21 total)
  fvarMetadata: Column Description labelDescription
experimentData: use 'experimentData(object)'
Annotation: GPL3533 


$ppbmc3
$ppbmc3$GSE30874_series_matrix.txt.gz
ExpressionSet (storageMode: lockedEnvironment)
assayData: 24123 features, 24 samples 
  element names: exprs 
protocolData: none
phenoData
  sampleNames: GSM765928 GSM765929 ... GSM765951 (24 total)
  varLabels: title geo_accession ... data_row_count (36 total)
  varMetadata: labelDescription
featureData
  featureNames: AFFX-BioB-3_at AFFX-BioB-5_at ... SscAffx.9.1.S1_at (24123 total)
  fvarLabels: ID Gene title ... GO:Component ID (21 total)
  fvarMetadata: Column Description labelDescription
experimentData: use 'experimentData(object)'
Annotation: GPL3533 


$hpbmc
$hpbmc$GSE13501_series_matrix.txt.gz
ExpressionSet (storageMode: lockedEnvironment)
assayData: 54675 features, 195 samples 
  element names: exprs 
protocolData: none
phenoData
  sampleNames: GSM340184 GSM340185 ... GSM340378 (195 total)
  varLabels: title geo_accession ... data_row_count (39 total)
  varMetadata: labelDescription
featureData
  featureNames: 1007_s_at 1053_at ... AFFX-TrpnX-M_at (54675 total)
  fvarLabels: ID Gene title ... GO:Component ID (21 total)
  fvarMetadata: Column Description labelDescription
experimentData: use 'experimentData(object)'
Annotation: GPL570 


$hlungtx1
$hlungtx1$GSE8021_series_matrix.txt.gz
ExpressionSet (storageMode: lockedEnvironment)
assayData: 17635 features, 50 samples 
  element names: exprs 
protocolData: none
phenoData
  sampleNames: GSM198616 GSM198617 ... GSM198665 (50 total)
  varLabels: title geo_accession ... data_row_count (32 total)
  varMetadata: labelDescription
featureData
  featureNames: AFFX-BioB-3_at AFFX-BioB-5_at ... XR_001531_at (17635 total)
  fvarLabels: ID Gene title ... GO:Component ID (21 total)
  fvarMetadata: Column Description labelDescription
experimentData: use 'experimentData(object)'
Annotation: GPL5356 


$hlungtx2
$hlungtx2$GSE9102_series_matrix.txt.gz
ExpressionSet (storageMode: lockedEnvironment)
assayData: 18746 features, 26 samples 
  element names: exprs 
protocolData: none
phenoData
  sampleNames: GSM230361 GSM230362 ... GSM230386 (26 total)
  varLabels: title geo_accession ... data_row_count (43 total)
  varMetadata: labelDescription
featureData
  featureNames: 1 2 ... 18746 (18746 total)
  fvarLabels: ID GB_ACC Name SPOT_ID
  fvarMetadata: Column Description labelDescription
experimentData: use 'experimentData(object)'
Annotation: GPL5909 


$hlavtx
$hlavtx$GSE2018_series_matrix.txt.gz
ExpressionSet (storageMode: lockedEnvironment)
assayData: 22283 features, 34 samples 
  element names: exprs 
protocolData: none
phenoData
  sampleNames: GSM36423 GSM36424 ... GSM36456 (34 total)
  varLabels: title geo_accession ... data_row_count (27 total)
  varMetadata: labelDescription
featureData
  featureNames: 1007_s_at 1053_at ... AFFX-TrpnX-M_at (22283 total)
  fvarLabels: ID Gene title ... GO:Component ID (21 total)
  fvarMetadata: Column Description labelDescription
experimentData: use 'experimentData(object)'
Annotation: GPL96 

gsmlists[['pig lung']] <- gsmlists[['pig lung']][
  extSubVar(gsmlists[['pig lung']], 13, 4) == "Uninfected"
  ]
gsmlists[['pig lung']] <- gsmlists[['pig lung']][
  str_detect(selVar(gsmlists[['pig lung']], 2),"Lung ")
  ]
gsmlists[['pig lung 2']] <- c("GSM18790", "GSM18791", "GSM18949", "GSM18950")
gsmlists[['pig lung 3']] <- morepl[[7]][extSubVar(morepl[[7]], 13, 2) == "none"]
gsmlists[['pig lung 4']] <- c("GSM43176", "GSM43179", "GSM43183", "GSM43178", "GSM43181", "GSM43182")
